### ISSUE TITLE
Missing Test Case - 472. Concatenated Words

### ISSUE THREAD

**Issue by DavidGMdev** [LeetCode-Feedback/LeetCode-Feedback#11381]
#### Category of the bug
- [ ] Question
- [ ] Solution
- [ ] Language
- [x] Missing Test Cases 

#### Description of the bug

// BUG : Naive recursive solution passes with AC verdict 

Testcase : 
["a","aa","aaa","aaaa","aaaaa","aaaaaa","aaaaaaa","aaaaaaaa","aaaaaaaaa","aaaaaaaaaa","aaaaaaaaaaa",
"aaaaaaaaaaaa","aaaaaaaaaaaaa","aaaaaaaaaaaaaa","aaaaaaaaaaaaaaa","aaaaaaaaaaaaaaaa",
"aaaaaaaaaaaaaaaaa","aaaaaaaaaaaaaaaaaa","aaaaaaaaaaaaaaaaaaa","aaaaaaaaaaaaaaaaaaaa",
"aaaaaaaaaaaaaaaaaaaaa","aaaaaaaaaaaaaaaaaaaaaa","aaaaaaaaaaaaaaaaaaaaaaa",
"aaaaaaaaaaaaaaaaaaaaaaaa","aaaaaaaaaaaaaaaaaaaaaaaaa"]

this testcase on the given code below must give "TLE" verdict , as the time complexity doesn't fit for doing 
such massive recursive calls results massive number of operations in worst case ( 2^0 + 2^1 + 2^3 + .....2^30 ) 
but the provided code pass with AC verdict so it think this is a missing edge case to be in consideration. 

Big thanks to [bemwaMalak](https://github.com/BemwaMalak) ❤️, 
for helping me calculating worst time complexity and discover this bug.
Leetcode username : bemwa 

#### Code you used for Submit/Run operation

```
class Solution {
public:
    map<string,bool> vis;
    vector<string> ans;
    void rec( string & x , string part , int i , bool & valid ){
        if ( i >= x.size() ) { 
            if ( part == "" || ( vis[part] && part != x ) ) valid = true ; 
            return;
        }

        // append
        string nxt = part;
        nxt.push_back(x[i]);
        rec(x,nxt,i+1,valid);

        // start new 
        // only if the prev are correct 
        if ( vis[part] ){
            string nxt = "";
            nxt.push_back(x[i]);
            rec(x,nxt,i+1,valid);
        }
        
    }
    vector<string> findAllConcatenatedWordsInADict(vector<string>& words) {
        for( int i = 0 ; i < words.size() ; i++ ) vis[words[i]] = true ;
        for( int i = 0 ; i < words.size() ; i++ ){
            bool valid = false ; 
            rec(words[i],"",0,valid);
            if (valid) ans.push_back(words[i]) ;
        }
        return ans; 
    }
};
```

#### Additional context

Expected Behavior : TLE verdict
Happens : AC verdict

**Comment by LC-Pam** [comment]
Thank you for reaching out to us, the above code received TLE after we submitted it. 

This could occur because the code is not optimized. When you run your code by "Submit", it's very different from running it with "Run Code". "Submit" run all of the test cases for this problem at once, whereas "Run Code" only run one test case at a time. Furthermore, getting Time Limit Exceeded from "Submit" does not mean that your code is not passing a certain test case; it simply means that your code is taking too long to run all of the test cases at once. I found a Swift solution in the problem discussion section and hope it can provide some inspiration to how your code can be improved to overcome this TLE issue: https://leetcode.com/problems/elimination-game/discuss/245342/Swift-keep-left-and-right-boundary

The issue you experienced is runtime variation. It normally happens when your code is at the edge of the time limit that can pass the problem. Therefore, your code may have different results at different submissions.
 It’s very difficult and ongoing research to evaluate the program’s runtime and get the same value for different runs of the same program. We have tried our best to make sure we isolate your program’s runtime, but there exist a lot of factors that make different executions have runtime variations.
 
I've passed your feedback to our engineering team and they'll investigate in order to provide consistent results for our users. However, I'm afraid that we won't have a quick fix for the issue that you're facing. I recommend that you optimize your code to pass the time limit. We apologize for the inconvenience and thank you for your support!
